netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -186,6 +185,12 @@ private void onRstStreamRead(int streamId)
     stream.transportReportStatus(Status.UNKNOWN, false, new Metadata.Trailers());
   }
 
+  @Override
+  public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exception {
+    goAwayStatus(Status.UNAVAILABLE.withDescription("Network channel closed by the client"));
```

Comment:
It will provide a better error message indicating that the client application initiated shutdown.  That is all I'm looking to accomplish here.  It's not going to mask any real error messages.